### ISSUE TITLE
Feature request: Use feature columns for labels

### ISSUE THREAD

**Issue by Baschdl** [tensorflow/tensorflow#18469]
As you describe in the documentation
> Think of feature columns as the intermediaries between raw data and Estimators. Feature columns are very rich, enabling you to transform a diverse range of raw data into formats that Estimators can use, allowing easy experimentation. [https://www.tensorflow.org/get_started/feature_columns](https://www.tensorflow.org/get_started/feature_columns)

feature columns are pretty handy to transform input data. I would like to see the same functionality for labels. Why can't I transform labels with the same "data columns" as features?

Have I written custom code N/A
OS Platform and Distribution N/A
TensorFlow installed from N/A
TensorFlow version N/A
Bazel version N/A
CUDA/cuDNN version N/A
GPU model and memory N/A
Exact command to reproduce N/A

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by Baschdl** [comment]
@tensorflowbutler Updated

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ispirmustafa** [comment]
Hi @Baschdl 
Most of the time complexity is around features. That's why we designed FeatureColumn. 
For labels, most of the time main transformation is converting strings to indices (classification). For that we added 'vocabulary_keys' to pre-made estimators such as [DNNClassifier](https://www.tensorflow.org/api_docs/python/tf/estimator/DNNClassifier). 
For other kind of transformations I would recommend to do that as part of `input_fn`
